### PR TITLE
Refactor situations/charts.py

### PR DESCRIPTION
**Pull request by MaxGhenis**
* Add type hinting and docstrings
* Create single metadata DataFrame for each variable
* Remove legend titles
* A lot of other refactoring

One of the last commits broke the waterfall chart, which now looks like this:
image

image

@nikhilwoodruff anything pop out here? Otherwise I'll take a look at what the dataframe currently looks like before it goes into the chart and get it back looking like it did.

TODO: Undo url changes

### CODE REVIEW

File: server/situations/charts.py [hunk 1]
```diff
@@ -69,72 +102,85 @@ def mtr_chart(baseline, reformed):
...
+COMPONENTS.loc["self_employment_income"] = [True, "Self-employment income"]
+COMPONENTS.loc["pension_income"] = [True, "Pension income"]
+COMPONENTS.loc["savings_interest_income"] = [True, "Savings income"]
+COMPONENTS.loc["dividend_income"] = [True, "Dividend income"]
+COMPONENTS.loc["income_tax"] = [False, "Income tax"]
+COMPONENTS.loc["national_insurance"] = [False, "National Insurance"]
+COMPONENTS.loc["universal_credit"] = [True, "Universal Credit"]
+COMPONENTS.loc["child_benefit"] = [True, "Child Benefit"]
+COMPONENTS.loc["UBI"] = [True, "Basic income"]
+COMPONENTS.loc["net_income"] = [True, "Net income"]
```
Comment (nikhilwoodruff) [inline server/situations/charts.py:124]: Thanks for doing this here: I added this here to get it working, though these might be good to eventually incorporate into the variable metadata in openfisca-uk, including making the label attribute more consistent and using that. I filed [#161](https://github.com/PSLmodels/openfisca-uk/issues/161) and [#162](https://github.com/PSLmodels/openfisca-uk/issues/162) to keep track of this

**Review by nikhilwoodruff — COMMENTED** [review]
Thanks @MaxGhenis! Looking much more readable.

**Comment by nikhilwoodruff** [comment]
FWIW I couldn't find out exactly what caused the Plotly formatting issue, and there's been a few PRs involving this file, so merging those might be easier (I can have another go when I get a chance)

**Comment by MaxGhenis** [comment]
This got too out of date, so I started over in PolicyEngine/policyengine-uk#207

**Pull request closed without merging**